### PR TITLE
Add docs for allocation profiling

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 4cdc3c39da0edf6f5115249ca4741ef14dc3b5fb with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58810/4cdc3c39da0edf6f5115249ca4741ef14dc3b5fb/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58810/4cdc3c39da0edf6f5115249ca4741ef14dc3b5fb/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: docs/en/operations/allocation-profiling.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+For example:
+
+- if we want to generate a text file with each procedure written per line:
+
+```sh
+jeprof path/to/binary path/to/heap/profile --text > result.txt
+```
+
+- if we want to generate a pdf file with call-graph:
```
Comment (antaljanosbenjamin) [inline docs/en/operations/allocation-profiling.md:92]:
```suggestion
- if we want to generate a PDF file with call-graph:
```

File: docs/en/operations/allocation-profiling.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+</TabItem>
+</Tabs>
+
+It's also possible to control the initial state of the profiler by setting `prof_active` option which is enabled by default.  
+For example, if we don't want to sample allocations during startup but only after we enable the profiler, we can start ClickHouse/Keeper with following environment variable:
+```sh
+MALLOC_CONF=background_thread:true,prof:true,prof_active:false
+```
+
+and enable profiler at later point.
```
Comment (antaljanosbenjamin) [inline docs/en/operations/allocation-profiling.md:159]:
```suggestion
and enable profiler at a later point.
```

File: docs/en/operations/allocation-profiling.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+## Additional options for profiler
+
+`jemalloc` has many different options available related to profiler which can be controlled by modifying `MALLOC_CONF` environment variable.
+For example, interval between allocation samples can be controlled with `lg_prof_sample`.  
+If you want to dump heap profile every N bytes you can enable it using `lg_prof_interval`.  
+
+We recommend to check `jemalloc`s [reference page](https://jemalloc.net/jemalloc.3.html) for such options.
+
+## Other resources
```
Comment (antaljanosbenjamin) [inline docs/en/operations/allocation-profiling.md:169]: Add some note about these sources might not correlate perfectly with jemalloc stats.

**Review by antaljanosbenjamin — APPROVED** [review]

**Pull request merged**